src/Compilers/Core/Portable/Symbols/TypeCompareKind.cs

```diff
@@ -19,7 +19,14 @@ internal enum TypeCompareKind
         IgnoreNullableModifiersForReferenceTypes = 8,
         UnknownNullableModifierMatchesAny = 16,
 
-        AllNullableIgnoreOptions = IgnoreNullableModifiersForReferenceTypes | UnknownNullableModifierMatchesAny,
+        /// <summary>
+        /// Different flavors of Nullable are equivalent,
+        /// different flavors of Not-Nullable are equivalent unless the type is possibly nullable refernce type parameter.
+        /// This option can cause cycles in binding if used too early!
+        /// </summary>
+        IgnoreInsignificantNullableModifiersDifference = 32,
```

Comment:
>IgnoreInsignificantNullableModifiersDifference [](start = 8, length = 46)

The PR adds a few places that use `IgnoreInsignificantNullableModifiersDifference`, but it is not so obvious what are the places that care about such differences.
Could you give an example or scenario where insignificant differences matter?
Should those places get a comment saying "we care about insignificant nullability differences here because ..."?

Reply:
There are only tree consumers of this kind (I will add comments in them):
- ConversionBase.HasIdentityConversionInternal
- two overloads of NullableWalker.IsNullabilityMismatch


---
In reply to: [232028709](https://github.com/dotnet/roslyn/pull/30913#discussion_r232028709) [](ancestors = 232028709)